pkg/relayer/proxy/sync.go

```diff
@@ -375,6 +389,48 @@ func (server *relayMinerHTTPServer) serveSyncRequest(
 	return relayRequest, nil
 }
 
+// serviceConfigTypeDefault indicates that the service config being used is
+// the default service config, as opposed to an RPC-type specific config.
+const serviceConfigTypeDefault = "DEFAULT_SERVICE_CONFIG"
+
+// getServiceConfig returns the service config for the service.
+// This will use either the RPC type specific service config or the default service config.
+func getServiceConfig(
+	supplierConfig *config.RelayMinerSupplierConfig,
+	request *http.Request,
+) (
+	serviceConfig *config.RelayMinerSupplierServiceConfig,
+	serviceConfigTypeLog string,
+	err error,
+) {
+	// If the following are true:
+	// 	- The RPC type is set for the service
+	// 	- The RPC type specific service config is available
+	// Then, use the RPC type specific service config.
+	rpcTypeHeaderValue := request.Header.Get(RPCTypeHeader)
+
+	if rpcTypeHeaderValue != "" {
+		// Attempt to convert string header value to int32.
+		// For example, "1" -> RPCType_GRPC, "2" -> RPCType_WEBSOCKET, etc.
+		rpcTypeInt, err := strconv.Atoi(rpcTypeHeaderValue)
+		if err != nil {
+			return nil, "", ErrRelayerProxyInternalError.Wrapf(
+				"unable to parse rpc type header value %q",
+				rpcTypeHeaderValue,
+			)
+		}
+
+		// If the header is successfully parsed, use the RPC type specific service config.
+		rpcType := sharedtypes.RPCType(rpcTypeInt)
+		if rpcTypeServiceConfig, ok := supplierConfig.RPCTypeServiceConfigs[rpcType]; ok {
+			return rpcTypeServiceConfig, fmt.Sprintf("%s_SERVICE_CONFIG", rpcType.String()), nil
+		}
+	}
+
```

Comment:
Please add an error line here with some details about the request (service-id, rpc-type, etc...) explaining we need to default.

It'll be crucial in debugging in the future.

Reply:
This is not technically an error. It just indicates either that the `Rpc-Type` header was provided with an invalid value (should not happen as this is handled by PATH) **or** the service is not configured with RPC type service configs.

I've added logs to explain all of these cases, including emojis.